neurodsp/rhythm/swm.py

```diff
@@ -19,20 +19,20 @@ def sliding_window_matching(sig, fs, win_len, win_spacing, max_iterations=500,
     fs : float
         Sampling rate, in Hz.
     win_len : float
-        Window length, in seconds.
+        Window length, in seconds. This is L in the original paper.
     win_spacing : float
-        Minimum window spacing, in seconds.
-    max_iterations : int
+        Minimum spacing between windows, in seconds. This is G in the original paper.
+    max_iterations : int, optional, default: 500
         Maximum number of iterations of potential changes in window placement.
-    temperature : float
-        Temperature parameter. Controls probability of accepting a new window.
+    temperature : float, optional, default: 1
+        Controls the probability of accepting a new window.
```

Comment:
This is a totally optional extension - if you want: but I do find it a little weird that "win_spacing" generally determines the number of windows. In the original, there is an option to set a specific number of windows. 
What do you think about extending `window_starts_custom` to not only take an array (as current), but also take an int, and if so, that is the the number of windows that are used (that get initialized as evenly spaced). I think this would be easy to add?

Reply:
This would basically be the same as using:
```
win_spacing=len(sig)/n_spaces
```
where `n_spaces` it the int you wanted to add as an optional param), right?

Reply:
Yeh, instead of overloading `window_starts_custom`, an optional parameter of `n_spaces` could be added (though I think I'd say this could / should be called `n_windows`). In the case of passing in this parameter, then `win_spacing` wouldn't have to be explicitly set, and could be computed as you have above. 

At least, that's how I remember it from the original implementation. The Matlab code might have a different API for setting these things (which I don't think we need to explicitly follow) - but the option to set the number of windows is something that is described in # 2 of the algorithm description in the paper.